### ISSUE TITLE
Azure Vault Auditing Disabled should not generate false positive if data source is absent

### ISSUE THREAD

**Issue by t3mi** [Checkmarx/kics#3285]
Azure _Vault Auditing Disabled_ query checks for presence of the data source for the same key vault (not getting why) and produces false positive if data source is absent.

### Expected Behavior
No issue.
### Actual Behavior
```
"line": 52,
"issue_type": "MissingAttribute",
"search_key": "azurerm_key_vault[kv]",
"search_value": "",
"expected_value": "'azurerm_monitor_diagnostic_setting' exists",
"actual_value": "'azurerm_monitor_diagnostic_setting' is missing",
"value": null
```
### Steps to Reproduce the Problem
Configuration
```hcl
resource "azurerm_key_vault" "kv" {
  name                            = "veryuniquekv"
  location                        = azurerm_resource_group.rg.location
  resource_group_name             = azurerm_resource_group.rg.name
  tenant_id                       = data.azurerm_client_config.current.tenant_id
  sku_name                        = "standard"
  enabled_for_disk_encryption     = true
  enabled_for_deployment          = true
  enabled_for_template_deployment = true
  purge_protection_enabled        = false
}

resource "azurerm_monitor_diagnostic_setting" "kv" {
  name                       = "AuditLogs"
  target_resource_id         = azurerm_key_vault.kv.id
  log_analytics_workspace_id = azurerm_log_analytics_workspace.oms.id
  
  log {
    category = "AuditEvent"

    retention_policy {
      enabled = false
    }
  }

  metric {
    category = "AllMetrics"

    retention_policy {
      enabled = false
    }
  }
}
```

### Specifications

-   Version: 1.3.0
-   Platform: Windows

**Comment by rafaela-soares** [comment]
Hello, @t3mi 🙂! Thank you for your report.
The query is verifying in any Terraform sample if the resource `azurerm_monitor_diagnostic_setting` is undefined or does not reference the resource 'azurerm_key_vault' in question and should not. It should check in the entire project. We will correct it as soon as possible!

**Comment by rafaela-soares** [comment]
https://github.com/Checkmarx/kics/blob/master/assets/queries/terraform/azure/vault_auditing_disabled/metadata.json
